### DISCUSSION TITLE
Increase DefaultMaxPerRoute value

### DISCUSSION THREAD

**Discussion by nxtruong72** [rubenlagus/TelegramBots#1360] [Q&A]
Hi,
I check in the code and see that we're using the default value for DefaultMaxPerRoute in the http client.
It means we only use 2 connections concurrently to the telegram server.

Is there any reason for that? And how can I setting that value?
I would like to increase the performance, because 2 connections concurrently was too small for me.

**Comment by Nuwan-Srimal** [comment]
The DefaultMaxPerRoute value in an HTTP client controls the maximum number of connections allowed per route, which essentially limits the number of concurrent connections to a specific server (in this case, the Telegram server). The default value of 2 is often set to ensure that applications do not overwhelm a server with too many simultaneous requests, which could lead to performance issues or even blocking by the server.

Reasons for Default Setting
Resource Management: Limiting connections helps manage system resources like memory and CPU, especially in applications that might be connecting to multiple servers simultaneously.
Fair Usage: It ensures fair usage of server resources, preventing a single client from hogging too many connections, which could lead to throttling or blocking by the server.
Stability: Having a lower number of concurrent connections can lead to a more stable and predictable behavior, especially under load.
